docs/developer-docs/web-apps/verifiable-credentials/relying-party.mdx

```diff
@@ -0,0 +1,272 @@
+  derivationOrigin: "<[optional] origin for delegated identity>",
+  windowOpenerFeatures: "<[optional] window opener config string>",
+});
+```
+
+[Learn more about the JavaScript SDK for VCs](https://github.com/dfinity/verifiable-credentials-sdk/tree/main/js-library).
+
+This code executes the following steps:
+
+- Opens a new window or tab with the identity provider (Internet Identity (II)).
+
+- Waits for a window post message from II.
+
+- Sends a request to II through the window post message.
+
+- Waits for the response from the II.
+
+- Calls an `onSuccess` callback if the process was successful, but that does not necessarily mean that a credential was received.
+
+- Calls an `onError` callback if the process has some technical error, if the user closes the II window, or if II times out.
+
+Learn more about [how this workflow works](how-it-works.mdx).
+
+### Step 3: Receive a response
+
+If the JSON RPC request is successfully sent and received by the issuer API, Internet Identity will return the following JSON RPC response to the relying party:
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "verifiablePresentation": "eyJQ..."
+  }
+}
+```
+
+In this response, `verifiablePresentation`is a JWT based verifiable presentation (VP) containing two credentials:
+
+1. A verifiable credential issued by Internet Identity that relates the `credentialSubject` to the user principal.
+
+2. A verifiable credential issued by the issuer to the user principal.
+
+The order of the credentials in the VP should match the order above, such that the credential relating to the `credentialSubject` to the user principal should come first.
+
+[View an example of a verifiable presentation](https://www.w3.org/TR/vc-data-model/#example-jwt-payload-of-a-jwt-based-verifiable-presentation-non-normative) for reference.
+
+The relying party should not expect to receive an immediate response, as it should wait for one of the following events:
+
+- II sends a JSON-RPC response as described above.
+
+- II sends a JSON-RPC error response as described below.
+
+- The user closes the II window. This should be treated as an error.
+
+The relying party may also close the II window after some timeout. The user would then be notified by the relying party that the request failed.
+
+### Error messages
+
+If the workflow fails for any reason, Internet Identity will return an error. The error does not provide details on the cause of the failure to provide privacy protection.
+
+An example error can be found below:
+
+``` json
+{
+    "id": 1,
+    "jsonrpc": "2.0",
+    "error": {
+        "version": "1",
+        "code": "UNKNOWN"
+    }
+}
+```
+
+## Verifying the credentials
+
+To verify that the credentials received are valid, the returned JWT should contain a property in `credentialSubject` named `credentialType` that contains key-value entries that list the expected arguments for the credential type. For example, if the following credential is requested:
```

Comment:
Not exactly. This part of the document is more best practices on how to build credentials, not about verifying them.

Which reminds me that we should add a section `## Recommended Conventions` in the issuer docs.

Instead:

> To verify the credentials the relying party needs to validate the JWT received from the identity provider, both cryptographically and semantically.

> The JWT should contain two verifiable credentials in the specified order:
> - An "Id alias" credential which links the effective subject of the credential to a temporary alias. This credential is signed by the identity provider.
> - The actual credential requested by the user. The subject of this credential is id_alias (the one before) and it's signed by the issuer.

> The verification should also check that the subject of the credential matches the subject of the first credential.

> To learn more about verifying credentials, check out the [demo relying party's code](https://github.com/dfinity/vc-playground/blob/b2f8cd5f5397b7b355b6503df5af3789834b79e9/rp/src/main.rs#L262) and the helper used from [Internet Identity repository](https://github.com/dfinity/internet-identity/blob/fd38b48f9fdf9085b4903d865a1f5e79ad24f81a/src/vc_util/src/lib.rs#L319).

Reply:
Revised per feedback